packages/hebao_v1/contracts/stores/SecurityStore.sol

```diff
@@ -128,23 +143,32 @@ contract SecurityStore is DataStore
     function getLock(address wallet)
         public
         view
-        returns (uint _lock, address _guardian)
+        returns (uint _lock, address _module)
     {
         _lock = uint(wallets[wallet].lock);
-        _guardian = wallets[wallet].locker;
+        _module = wallets[wallet].lockedBy;
     }
 
-    function setLock(address wallet, uint lock, address guardian)
+    function setLock(
+        address wallet,
+        uint    lock
+        )
         public
         onlyManager
     {
-        require(lock == 0 || lock > now, "INVALID_LOCK_TIME");
+        if (isLocked(wallet)) {
+            require(
```

Comment:
Maybe we should keep this check in the module. I'm thinking of a scenario like with the recovery in multiple steps. As a first step the wallet owner may lock the wallet using a single guardian because it's quick. After he has gathered enough signatures he would start the recovery which locks the wallet again, but with a much stricter unlock criteria. With the current logic this would be impossible because the user would have to unlock the wallet first just so he can start the recovery that locks it again (and this stricter lock would be really needed, so skipping the lock is not an option when the wallet is already locked). So I think we should make it possible for modules to overwrite locks in some cases (but not in all cases, otherwise after recovery is started with a majority the wallet is locked with a majority, we don't want to allow the case where the lock is overwritten by a single guardian in `LockModule` so it can also be unlocked again by a single guardian).

Reply:
Agreed.